### ISSUE TITLE
Deserialize MesosCluster object

### ISSUE THREAD

**Issue by frankscholten** [ContainerSolutions/minimesos#213]
# Background

The minimesos process is a short lived process: when you invoke a minimesos command the minimesos process starts, performs the command and then exits. The containers on which it operates generally live at least a few minimesos commands and probably longer. To keep track of the Mesos containers between minimesos invocations we use the following identifiers

* cluster ID
* unique ID
* Docker container ID 

The first two are assigned by minimesos and the third is assigned by Docker when the container starts. We store the cluster ID in the minimesos.cluster file.

# The problem

Certain parts of the code base use static methods that operate on the cluster ID instead of the MesosCluster object because the MesosCluster object is not available at that point. It is only available on cluster creation. 

This causes code duplication and makes the codebase less flexible.

See for example this method from MesosCluster

```
    public static String getContainerIp(String clusterId, String role) {
        Container container = getContainer(clusterId, role);
        if ( container != null ) {
            return DockerContainersUtil.getIpAddress( dockerClient, container.getId() );
        }
        return null;
    }
```

It would have been better if the method ``AbstractContainer.getIpAddress()`` was used. 

# Solution

Create a constructor ```MesosCluster(String clusterId)``` which recreates the MesosCluster object from the cluster ID. The cluster ID and the containers field should be filled in the same way as the original MesosCluster object. This means the containers are sorted by created date, since on ``start()`` the containers are started in that order.

# Implementation

To implement this we have to take care of the following:

* [X] [Reduce MesosCluster responsibilities](https://github.com/ContainerSolutions/minimesos/pull/214)
* [x] Implement container deserialization
* [x] Implement MesosCluster deserialization
* [x] Replace static methods

**Comment by frankscholten** [comment]
State and flags methods are now in Master container. See https://github.com/ContainerSolutions/minimesos/pull/214

**Comment by sadovnikov** [comment]
closed in #221
